pkg/operator/quorumguardcontroller/quorumguardcontroller.go

```diff
@@ -172,7 +173,45 @@ func (c *QuorumGuardController) ensureEtcdGuardDeployment(ctx context.Context, r
 	// use image from release payload
 	c.etcdQuorumGuard.Spec.Template.Spec.Containers[0].Image = c.cliImagePullSpec
 
-	// if restart occurred, we will apply etcd guard deployment but if it is the same, nothing will happened
+	affinity := &corev1.Affinity{
+		// Ensure only a single instance is deployed per node
+		PodAntiAffinity: &corev1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      "k8s-app",
+								Operator: metav1.LabelSelectorOpIn,
+								Values:   []string{"etcd-quorum-guard"},
+							},
+						},
+					},
+					TopologyKey: "kubernetes.io/hostname",
```

Comment:
You're probably reworking this but this is already present here:
https://github.com/openshift/cluster-etcd-operator/blob/83e4a83079cfffbd3d8048f02f6d0603cd00c15a/pkg/operator/etcd_assets/bindata.go#L966-L973